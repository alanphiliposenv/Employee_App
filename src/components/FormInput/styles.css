.text-form-input {
  display: flex;
  flex-direction: column;
}

.form-input {
  height: 45px;
  margin-top: 8px;
  background: #fafafa 0% 0% no-repeat padding-box;
  border: 1px solid #d9d9d9;
}

.form-input[type="text"][disabled] {
   background: #D1D1D1 0% 0% no-repeat padding-box;
   color: #8B8D9D;
}

.form-input-label {
    height: 19px;
    font-size: 16px;
}
