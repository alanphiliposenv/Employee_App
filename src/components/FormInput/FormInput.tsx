import { FC } from 'react';
import './styles.css';

type InputPropTypes = {
  type: 'text' | 'password';
  placeholder: string;
  label?: string;
  value: string;
  onChange: (v: string) => void;
  disabled?: boolean;
};

const FormInput: FC<InputPropTypes> = ({ type, placeholder, label, value, onChange, disabled }) => {
  return (
    <div className='text-form-input'>
      <label className='form-input-label'>{label}</label>
      <input
        disabled={disabled}
        className='form-input'
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={(e) => onChange(e.target.value)}
      />
    </div>
  );
};

export default FormInput;
