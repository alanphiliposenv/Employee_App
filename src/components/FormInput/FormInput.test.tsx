import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import FormInput, { InputPropTypes } from './FormInput';

describe('Tests for FormInput', () => {
  test('if type correctly rendered as text', () => {
    const FormInputProps: InputPropTypes = {
      type: 'text'
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).toHaveAttribute('type', 'text');
    expect(element).not.toHaveAttribute('type', 'password');
  });
  test('if type correctly rendered as password', () => {
    const FormInputProps: InputPropTypes = {
      type: 'password'
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).toHaveAttribute('type', 'password');
    expect(element).not.toHaveAttribute('type', 'text');
  });
  test('if label rendered correctly', () => {
    const FormInputProps: InputPropTypes = {
      label: 'label-text'
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-label');

    expect(element).toHaveTextContent('label-text');
    expect(element).not.toHaveTextContent('wrong-label');
  });
  test('if input not disabled when false', () => {
    const FormInputProps: InputPropTypes = {
      disabled: false
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).not.toHaveAttribute('disabled');
  });
  test('if input disabled when true', () => {
    const FormInputProps: InputPropTypes = {
      disabled: true
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).toHaveAttribute('disabled');
  });
  test('if placehold rendered correctly', () => {
    const FormInputProps: InputPropTypes = {
      placeholder: 'placeholder-text'
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).toHaveAttribute('placeholder', 'placeholder-text');
    expect(element).not.toHaveAttribute('placeholder', 'wrong-placeholder');
  });
  test('if value rendered correctly', () => {
    const FormInputProps: InputPropTypes = {
      value: 'value-text'
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).toHaveAttribute('value', 'value-text');
    expect(element).not.toHaveAttribute('value', 'wrong-value');
  });
  test('if onChangne rendered correctly', () => {
    const onChange = jest.fn();
    const FormInputProps: InputPropTypes = {
      value: 'value-text',
      onChange
    };

    render(<FormInput {...FormInputProps} />);
    const element = screen.getByTestId('form-input-field');

    expect(element).toHaveAttribute('value', 'value-text');
    fireEvent.change(element, { target: { value: 'new-value-text' } });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('new-value-text');
  });
  test('snapshot redered', () => {
    const FormInputProps: InputPropTypes = {
      value: 'value-text',
      placeholder: 'placeholder-text',
      label: 'label-text'
    };

    const element = render(<FormInput {...FormInputProps} />);

    expect(element).toMatchSnapshot();
  });
});
