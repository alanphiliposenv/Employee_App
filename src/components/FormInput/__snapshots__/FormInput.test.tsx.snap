// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for FormInput snapshot redered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-form-input"
      >
        <label
          class="form-input-label"
          data-testid="form-input-label"
        >
          label-text
        </label>
        <input
          class="form-input"
          data-testid="form-input-field"
          placeholder="placeholder-text"
          value="value-text"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-form-input"
    >
      <label
        class="form-input-label"
        data-testid="form-input-label"
      >
        label-text
      </label>
      <input
        class="form-input"
        data-testid="form-input-field"
        placeholder="placeholder-text"
        value="value-text"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
